app/src/main/java/org/mifos/mobile/ui/savings_account_application/SavingsAccountApplicationViewModel.kt

```diff
@@ -0,0 +1,128 @@
+package org.mifos.mobile.ui.savings_account_application
+
+import androidx.compose.runtime.State
+import androidx.compose.runtime.mutableStateOf
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.catch
+import kotlinx.coroutines.launch
+import org.mifos.mobile.R
+import org.mifos.mobile.api.local.PreferencesHelper
+import org.mifos.mobile.models.accounts.savings.SavingsAccountApplicationPayload
+import org.mifos.mobile.models.accounts.savings.SavingsAccountUpdatePayload
+import org.mifos.mobile.models.accounts.savings.SavingsWithAssociations
+import org.mifos.mobile.models.templates.savings.SavingsAccountTemplate
+import org.mifos.mobile.repositories.SavingsAccountRepository
+import org.mifos.mobile.ui.enums.SavingsAccountState
+import org.mifos.mobile.utils.DateHelper
+import org.mifos.mobile.utils.getTodayFormatted
+import javax.inject.Inject
+
+@HiltViewModel
+class SavingsAccountApplicationViewModel @Inject constructor(private val savingsAccountRepositoryImp: SavingsAccountRepository) :
+    ViewModel() {
+
+    @Inject
+    lateinit var preferencesHelper: PreferencesHelper
```

Comment:
You don't neet to inject here, you can directly inject in constructor like 

@HiltViewModel
class SavingsAccountApplicationViewModel @Inject constructor(
private val savingsAccountRepositoryImp: SavingsAccountRepository, 
val preferencesHelper: PreferencesHelper
) : ViewModel() {

Reply:
oh sorry, i missed that. Updated now